java-sdk/src/main/java/com/adobe/target/artifact/ArtifactObfuscator.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * ADOBE CONFIDENTIAL
+ * __________________
+ *
+ * Copyright 2020 Adobe Systems Incorporated
+ * All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Adobe Systems Incorporated and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Adobe Systems Incorporated and its
+ * suppliers and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Adobe Systems Incorporated.
+ */
+package com.adobe.target.artifact;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+public class ArtifactObfuscator {
+
+  private static final int RANDOM_KEY_SIZE = 32;
+  private static final byte[] HEADER_VERSION = "ATOD:001".getBytes(StandardCharsets.UTF_8);
+
+  public String unobfuscate(String obfuscationKey, byte[] content) throws TargetInvalidArtifactException {
+    validateContent(content);
+
+    byte[] firstPart = obfuscationKey.getBytes(StandardCharsets.UTF_8);
+    byte[] secondPart = extractRandomKey(content);
+    byte[] key = buildKey(firstPart, secondPart);
+
+    byte[] obfuscatedArtifact = extractObfuscatedArtifact(content);
```

Comment:
This seems like an unnecessary copying of a large byte array. You should be able to xor the original obfuscated byte array and just skip the header without having to do a full copy first.

Reply:
I tried to skip the header during xor but I ended up having an extra add operation for each byte processed in the loop because I have to keep track where I read and where I write.  In the this, moving a block of memory might be better than numerous of add operations.

Reply:
You should only have to offset the key into the content array by HEADER_VERSION.length + RANDOM_KEY_SIZE everytime. Add operations are way cheaper than copying hundreds of KB of memory unnecessarily.

Reply:
Fixed.